a10_octavia/controller/worker/tasks/a10_compute_tasks.py

```diff
@@ -92,20 +93,25 @@ class ValidateComputeForProject(BaseComputeTask):
 
     def execute(self, loadbalancer, role):
         self.vthunder_repo = a10_repo.VThunderRepository()
+        self.amphora = repo.AmphoraRepository()
+        self.loadbalancer_repo = a10_repo.LoadBalancerRepository()
         vthunder_ids = self.vthunder_repo.get_vthunders_by_project_id_and_role(
             db_apis.get_session(), loadbalancer.project_id, role)
         for vthunder_id in vthunder_ids:
             vthunder = self.vthunder_repo.get(
                 db_apis.get_session(),
                 id=vthunder_id)
-            try:
-                amp, fault = self.compute.get_amphora(vthunder.compute_id)
-                if amp.compute_id == vthunder.compute_id:
-                    LOG.debug("Successfully validated comput_id %s for the project %s",
-                              vthunder.compute_id, vthunder.project_id)
-                    break
-            except Exception:
-                pass
+            lb = self.loadbalancer_repo.get_lb_excluding_deleted(
+                db_apis.get_session(), vthunder.loadbalancer_id)
+            if lb:
+                amphora = self.amphora.get(db_apis.get_session(), load_balancer_id=lb.id)
+                try:
+                    if amphora.compute_id == vthunder.compute_id:
+                        LOG.debug("Successfully validated comput_id %s for the project %s",
+                                  vthunder.compute_id, vthunder.project_id)
+                        break
+                except Exception:
+                    pass
```

Comment:
it seems that, this Exception handling i.e pass Exception will not be needed here now

Reply:
Removed try-except block as it is not needed now and shifted ValidateComputeForProject to a10_database_tasks